packages/block-library/src/search/index.php

```diff
@@ -142,6 +139,12 @@ function render_block_core_search( $attributes ) {
 		array( 'class' => $classnames )
 	);
 
+	if ( ! empty( $attributes['buttonPosition'] ) && ! empty( $attributes['buttonBehavior'] ) ) {
+		if ( 'button-only' === $attributes['buttonPosition'] && 'expand-searchfield' === $attributes['buttonBehavior'] ) {
+			wp_enqueue_script( 'wp-block--search-view', plugins_url( 'search/view.min.js', __FILE__ ) );
```

Comment:
Do we need to enqueue the `view.min.js` file here? This seems to be triggering multiple event listeners for me when I click the search button:

<img width="1036" alt="image" src="https://github.com/WordPress/gutenberg/assets/1645628/9208e94b-8864-4f11-b25e-ed566eafbb63">

If I remove this line, `toggleSearchField` is only triggered once.

Although I know there's additional logic for its inclusion here (`button-only` and `expand-searchfield`), so maybe `view.min.js` shouldn't be included in block.json? I'm not sure which is the best, but I think it should only be included once. It's probably better if it's removed from block.json and kept here (in search/index.php), as then it's only included when the functionality is required.

Reply:
Oh good call! I removed it from block.json [73176c1](https://github.com/WordPress/gutenberg/pull/50487/commits/73176c118e8098413b5880557930b4a57e1446a6). I thought it was needed to enqueue the file, but I guess that ends up loading it twice.